src/main/java/seedu/intrack/logic/parser/TaskCommandParser.java

```diff
@@ -0,0 +1,55 @@
+package seedu.intrack.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.intrack.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.intrack.commons.core.index.Index;
+import seedu.intrack.commons.exceptions.IllegalValueException;
+import seedu.intrack.logic.commands.TaskCommand;
+import seedu.intrack.logic.parser.exceptions.ParseException;
+import seedu.intrack.model.internship.Task;
+
+/**
+ * Parses input arguments and creates a new {@code TaskCommand} object
+ */
+public class TaskCommandParser implements Parser<TaskCommand> {
+    private static final Pattern TASK_COMMAND_FORMAT =
+            Pattern.compile("(?<index>\\d+)\\s+(?<description>.*)\\s+/at(?<dateTime>.*)");
```

Comment:
Would be good to provide a comment explaining what this regex means so other collaborators can make use of it more easily.

Reply:
Excellent suggestion, I have already added it in now.